src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -168,52 +168,18 @@ async def _func(data: t.Any) -> t.Any:
 
         return _func
 
-    async def controller(self):
-        """
-        A standalone coroutine to wait/dispatch calling.
-        """
-        logger.debug("Starting dispatcher optimizer training...")
-        # warm up the model
-        while self.optimizer.outbound_counter <= self.optimizer.N_SKIPPED_SAMPLE:
-            try:
-                async with self._wake_event:  # block until there's any request in queue
-                    await self._wake_event.wait_for(self._queue.__len__)
-
-                n = len(self._queue)
-                now = time.time()
-                w0 = now - self._queue[0][0]
-
-                # only cancel requests if there are more than enough for training
-                if (
-                    n
-                    > self.optimizer.N_SKIPPED_SAMPLE
-                    - self.optimizer.outbound_counter
-                    + 6
-                    and w0 >= self.max_latency_in_ms
-                ):
-                    # we're being very conservative and only canceling requests if they have already timed out
-                    self._queue.popleft()[2].cancel()
-                    continue
-                # don't try to be smart here, just serve the first few requests
-                if self._sema.is_locked():
-                    await asyncio.sleep(self.tick_interval)
-                    continue
-
-                n_call_out = 1
-                # call
-                self._sema.acquire()
-                inputs_info = tuple(self._queue.pop() for _ in range(n_call_out))
-                self._loop.create_task(self.outbound_call(inputs_info))
-            except asyncio.CancelledError:
-                return
-            except Exception as e:  # pylint: disable=broad-except
-                logger.error(traceback.format_exc(), exc_info=e)
+    async def train_optimizer(
+        self, num_required_reqs: int, num_reqs_to_train: int, batch_size: int, wait: 0,
```

Comment:
```suggestion
        self, num_required_reqs: int, num_reqs_to_train: int, batch_size: int, wait: int = 0,
```
I think this is what you mean?

Reply:
Oh, no, I just forgot to remove that argument.